letsencrypt/scripts/main.py

```diff
@@ -9,61 +9,100 @@
+        help="Path to the private key file for certificate generation.")
+    add("-B", "--rsa-key-size", type=int, default=2048,
+        metavar="N", help="RSA key shall be sized N bits.")
+
+    add("-k", "--revoke", action="store_true", help="Revoke a certificate.")
+    add("-b", "--rollback", type=int, default=0, metavar="N",
+        help="Revert configuration N number of checkpoints.")
+    add("-v", "--view-config-changes", action="store_true",
+        help="View checkpoints and associated configuration changes.")
+    add("-r", "--redirect", action="store_true",
+        help="Automatically redirect all HTTP traffic to HTTPS for the newly "
+             "authenticated vhost.")
+
+    add("-e", "--agree-tos", dest="eula", action="store_true",
+        help="Skip the end user license agreement screen.")
+    add("-t", "--text", dest="use_curses", action="store_false",
+        help="Use the text output instead of the curses UI.")
+    add("--test", action="store_true", help="Run in test mode.")
+
+    # TODO: trailing slashes might be important! check and remove
+    add("--config-dir", default="/etc/letsencrypt/",
+        help="Configuration directory.")
+    add("--work-dir", default="/var/lib/letsencrypt/",
+        help="Working directory.")
+    add("--backup-dir", default="/var/lib/letsencrypt/backups/",
+        help="Configuration backups directory.")
+    add("--temp-checkpoint-dir",
+        default="/var/lib/letsencrypt/temp_checkpoint/",
+        help="Temporary checkpoint directory.")
+    add("--in-progress-dir",
+        default="/var/lib/letsencrypt/backups/IN_PROGRESS/",
+        help="Directory used before a permanent checkpoint is finalized")
+    add("--cert-key-backup", default="/var/lib/letsencrypt/keys-certs/",
+        help="Directory where all certificates and keys are stored. "
+             "Used for easy revocation.")
+    add("--rev-tokens-dir", default="/var/lib/letsencrypt/revocation_tokens/",
+        help="Directory where all revocation tokens are saved.")
+    add("--key-dir", default="/etc/letsencrypt/keys/", help="Keys storage.")
+    add("--cert-dir", default="/etc/letsencrypt/certs/",
+        help="Certificates storage.")
+
+    add("--le-vhost-ext", default="-le-ssl.conf",
+        help="SSL vhost configuration extension.")
+    add("--cert-path", default="/etc/letsencrypt/certs/cert-letsencrypt.pem",
+        help="Let's Encrypt certificate file.")
+    add("--chain-path", default="/etc/letsencrypt/certs/chain-letsencrypt.pem",
+        help="Let's Encrypt chain file.")
+
+    add("--apache-ctl", default="apache2ctl",
+        help="Path to the 'apache2ctl' binary, used for 'configtest' and "
+             "retrieving Apache2 version number.")
+    add("--apache-enmod", default="a2enmod",
+        help="Path to the Apache 'a2enmod' binary.")
+    add("--apache-init-script", default="/etc/init.d/apache2",
+        help="Path to the Apache init script (used for server reload/restart).")
```

Comment:
i think there need to be more thoughts about *who* may give pathes to binaries (or command lines) and *how*, see #221.

threat model:
user is unprivileged, has no root/sudo.
user may edit user configuration file (or commandline options for letsencrypt call).
letsencrypt is executed by some other tool/process with root/sudo (not the user, but provider tool, e.g.).

attack:
malicious user gives user config file / cmdline param pointing apache-init-script to an executable file he put into a writable directory on the server. provider tool runs the given executable as root.

not sure whether this threat model / use case is one you want to support (see #221), but if you want, there is a problem here.

maybe we need: admin config (done by root), user config (by user) and constants (optional, could be also in admin's config maybe). some stuff in the admin config could also get adapted by linux distributions, pointing stuff to correct places.